spec/requests/middleware_datasources_spec.rb

```diff
@@ -0,0 +1,68 @@
+describe 'Middleware Datasources API' do
+  let(:datasource) { FactoryGirl.create(:middleware_datasource) }
+
+  # For some reason middleware_datasources_url is not returning the full
+  # url, just the path portion. This is a hack, but will do the trick.
+  def datasource_url
+    id = ApplicationRecord.compress_id(datasource.id)
+    "http://www.example.com#{middleware_datasources_url(id)}"
+  end
+
+  def match_id
+    eq ApplicationRecord.compress_id(datasource.id).to_s
```

Comment:
You can do `datasource.compressed_id` here - which might obviate the need for this matcher.

Reply:
Oh, nice!